OPEN_PROBLEMS/HETEROGENEOUS_RUNTIMES.md

```diff
@@ -0,0 +1,190 @@
+# Heterogeneous Runtimes
+
+## Short Description
+Annual global IP traffic will reach 396 exabytes per month. The number of devices connected to the internet is expected to double in the same time. Access to high speed broadband connectivity is limited by poor last mile connections. Optic-fiber deployments are cost-prohibitive and slow to deploy.
+
+Edge  computing  has  emerged  as  a  distributed  computing paradigm to overcome practical scalability limits of cloud computing.  The main principle of edge computing is to leverage computational resources outside of the cloud to perform computations closer to data sources, avoiding unnecessary data transfers to the cloud and enabling faster responses for clients. Given the enormous amount of data that is expected to be produced at the edge of the network (by end-user devices), the edge-computing principle builds on the fact that “it is cheaper to bring computation to data, rather than data to computation”.
+
+Data storage and computation points at the edge of the network, be it in cell stations, WiFi access points, or even the same user devices that produce the data ultimately form a peer-to-peer network.
+
+*Both IPFS and libp2p can have a pivotal role in this new trend of distributed technologies and expansion of the Internet beyond traditional boundaries*. Both technologies already have implementations in several programming languages and are compatible for their execution in a great gamut of runtimes (desktop, browser, cloud, etc.) and network conditions (public IP addresses, behind NATs, etc.). Despite this success, there is still a lot of work to be done to make IPFS and libp2p compatible with any runtime and network condition in a way that offers a flawless experience, just like one expects when using the net or http package of any language. This open problem is built upon *the vision of making libp2p / IPFS modules and protocols the de-facto distributed substrate for connected devices in the near-future Internet*.
+
+At high-level, the aim of this research endeavor is to:
+-   Enable the execution of libp2p nodes and its underlying protocols anywhere (any device and any runtime).
+    - [Are we yet?](https://wiki.mozilla.org/Areweyet)
```

Comment:
```suggestion

-  Enable libp2p to leverage the available protocols from the multiple runtimes, so that it can execute and form a p2p network. This will require a coordinated approach to cover as much ground possible (possible approach inspired by [Are we yet?](https://wiki.mozilla.org/Areweyet) initiatives)
```